### PR TITLE
`fulfillRequests`: prefer using a ZIO while loop over a long chain of `flatMap`

### PR DESCRIPTION
**Pull request by guizmaii**
Reopening of this request based on @erikvanoosten and @adamgfraser feedback: https://github.com/zio/zio-kafka/pull/712

### CODE REVIEW

File: zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala [hunk 1]
```diff
@@ -224,7 +225,7 @@ private[consumer] final class Runloop(
     val buf = mutable.Map.empty[TopicPartition, Chunk[ByteArrayConsumerRecord]]
     buf ++= bufferedRecords.recs
 
-    var fulfillAction: UIO[_] = ZIO.unit
+    val fulfillActionsBuffer = ListBuffer.empty[UIO[Unit]]
```
Comment (erikvanoosten) [inline zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala:228]: With this suggestion the `asInstanceOf` on line 252 can be removed.
```suggestion
    val fulfillActionsBuffer = ListBuffer.empty[UIO[Boolean]]
```

**Review by erikvanoosten — DISMISSED** [review]
One small suggestion but otherwise LGTM!

**Comment by svroonland** [comment]
The changes in code look good to me. Could you post before and after benchmark results? (maybe wait until #736 is merged, which fixes some benchmark issues)

**Comment by guizmaii** [comment]
Not relevant anymore

**Pull request closed without merging**